src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -56,6 +57,13 @@ public class Http1OutputProducer : IHttpOutputProducer, IHttpOutputAborter, IDis
         private bool _currentChunkMemoryUpdated;
         private IMemoryOwner<byte> _fakeMemoryOwner;
 
+        // Fields needed to store writes before calling either startAsync or Write/FlushAsync
+        private LinkedList<CompletedBuffer> _completedSegments;
```

Comment:
Why a linked list?

Reply:
I'm not sure of the performance of a LinkedList in C#. But we don't need indexing, just iteration, so it seemed logical to use just a LinkedList?

Is it still better to use a List instead?

Reply:
Yes, just use a list. There's nothing here that requires a linked list.